### PR TITLE
feat: change default value of `checkOrigin`

### PR DESCRIPTION
**Pull request by ematipico**
## Changes

Closes PLT-2262

It changes the value of `security.checkOrigin` to `true`

## Testing

I updated the existing tests and removed the config. Tests should still pass

## Docs

N/A

/cc @withastro/maintainers-docs for feedback!

### CODE REVIEW

File: .changeset/itchy-toys-march.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+'astro': major
+---
+
+The option `security.checkOrigin` is `true` by default
```
Comment (sarah11918) [inline .changeset/itchy-toys-march.md:5]:
```suggestion
Updates the default value of `security.checkOrigin` to `true`, which enables Cross-Site Request Forgery (CSRF) protection by default for pages rendered on demand.
```

File: .changeset/itchy-toys-march.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+Update `astro.config.mjs` to _opt-out_:
+
+```diff
+export default defineConfig({
++  security: {
++    checkOrigin: false
++  }
+})
+```
```
Comment (sarah11918) [inline .changeset/itchy-toys-march.md:21]:
``````suggestion
If you had previously configured `security.checkOrigin: true`, you no longer need this set in your Astro config. This is now the default and it is safe to remove.

To disable this behavior and opt out of automatically checking that the “origin” header matches the URL sent by each request, you must explicitly set `security.checkOrigin: false`:

```diff
// astro.config.mjs
export default defineConfig({
+  security: {
+    checkOrigin: false
+  }
})
```
``````
This is one I had already mocked up in the upgrade guide, so we can use it!  :smile:

**Review by sarah11918 — COMMENTED** [review]
Thanks @ematipico ! This was one I already had started an entry for in the upgrade guide, so you can also [review the entry in the guide here](https://5-0-0-beta--astro-docs-2.netlify.app/en/guides/upgrade-to/v5/#changed-csrf-protection-is-now-set-by-default-plt2262) and adjust it as you see fit.

You will **also** need to update the `types/public/config.ts` file changing the default in the `@docs` entry to `true`.  I would also remove the "when enabled" from the start of the sentence and just start with "Performs a check...." since that's now the default behaviour.

That should probably be changed in this PR at the same time. (I think it's fine to keep the version number of the feature at 4.9 though, even though the default value has now changed.)

I like how you showed adding the opt-out, whereas my entry showed removing the 'true' setting as no longer needed. Either one is fine by me, whatever you prefer!

**Review by bluwy — APPROVED** [review]
Code-wise looks good 👍

**Review by matthewp — APPROVED** [review]

**Review by sarah11918 — APPROVED** [review]
Looks great @ematipico ! Approving for docs! :partying_face:

**Review by yanthomasdev — APPROVED** [review]
LGTM

**Pull request merged**